src/discord/commands/create_calendar.rs

```diff
@@ -0,0 +1,50 @@
+use crate::calendar::Client as CalendarClient;
+use serenity::all::{Context, CreateCommand, GuildId, Permissions, ResolvedOption};
+use tracing::{error, info, instrument};
+
+use crate::calendar::get_calendar_url;
+
+#[instrument]
+pub async fn run(ctx: &Context, guild_id: GuildId, _options: &[ResolvedOption<'_>]) -> String {
+    info!("Creating a calendar");
+    let lock = ctx.data.read().await;
+    let calendar_client = match lock
+        .get::<CalendarClient>() {
+            Some(client) => client,
+            None => {
+                error!("No calendar client found");
```

Comment:
https://doc.rust-lang.org/std/result/enum.Result.html#method.inspect_err
or 
https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err

I don't like the stringly typed error here.

Reply:
Fixed, now the commands return `MessageResult`